hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,105 @@
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class HrTimesheetSheet(models.Model):
+    _inherit = "hr_timesheet.sheet"
+
+    def _compute_attendance_count(self):
+        """Compute total number of attendance records
+        linked to current timesheet"""
+
+        for attendances in self:
+            attendances.attendance_count = len(attendances.attendances_ids)
+
+    @api.depends(
+        "timesheet_ids",
+        "timesheet_ids.unit_amount",
+        "attendances_ids",
+        "attendances_ids.check_in",
+        "attendances_ids.check_out",
+        "attendances_ids.employee_id",
+    )
+    def _compute_attendance_time(self):
+        """Compute total attendance time and
+        difference in total attendance-time
+        and timesheet-entry"""
+
+        current_date = datetime.now()
+        for sheet in self:
+            atte_without_checkout = sheet.attendances_ids.filtered(
+                lambda attendance: not attendance.check_out
+            )
+            atte_with_checkout = sheet.attendances_ids - atte_without_checkout
+            total_time = sum(atte_with_checkout.mapped("worked_hours"))
+            for attendance in atte_without_checkout:
+                delta = current_date - attendance.check_in
+                total_time += delta.total_seconds() / 3600.0
+            sheet.total_attendance = total_time
+
+            # calculate total difference
+            total_working_time = sum(sheet.mapped("timesheet_ids.unit_amount"))
+            sheet.total_difference = total_time - total_working_time
+
+    total_attendance = fields.Float(
+        compute="_compute_attendance_time",
+    )
+    total_difference = fields.Float(
+        compute="_compute_attendance_time",
+        string="Difference",
+    )
+    attendances_ids = fields.One2many(
+        comodel_name="hr.attendance", inverse_name="sheet_id", string="Attendances"
+    )
+    attendance_state = fields.Selection(
+        related="employee_id.attendance_state", string="Current Status"
```

Comment:
```suggestion
        related="employee_id.attendance_state", related_sudo=True, string="Current Status"
```
Without the sudo a user has no access to their attendance_state due to the `hr.employee.public` split, which leads to a missing field JavaScript error in this module for the use of `attendance_state` in another field's `invisible` attribute.

Reply:
Updated as per suggestion,
Thanks